ci_framework/roles/ci_network/tasks/main.yml

```diff
@@ -0,0 +1,65 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if environment file exists
+  register: env_file
+  ansible.builtin.stat:
+    path: "{{ cifmw_network_generated_layout }}"
+
+- name: Load generated network if exists
+  when:
+    - env_file.stat.exists
+  ansible.builtin.include_vars:
+    file: "{{  cifmw_network_generated_layout }}"
+
```

Comment:
One suggestion I have is to split out the valitation task from apply.yml into a new task file called something like validate.yml and call it after `Load generated network if exists`

Reply:
Maybe something to consider - as a follow-up though.